core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala

```diff
@@ -38,14 +40,27 @@ class OfferMatcher(masterDomainInfo: Mesos.DomainInfo) {
       result: Map[RunningPodSpec, List[OfferMatcher.ResourceMatch]],
       pendingLaunchPodSpecs: List[RunningPodSpec]): Map[RunningPodSpec, List[OfferMatcher.ResourceMatch]] = {
 
+    def matchesAgentFilter(agentFilters: Iterable[AgentFilter]): Boolean = {
+      agentFilters.find { filter =>
+        !filter(originalOffer)
+      } match {
+        case Some(nonMatchingfilter) =>
+          logger.debug(
+            s"Declining offer ${originalOffer.getId.getValue} for pod ${podSpec.id}; first non-matching agent filter: ${nonMatchingfilter.description}")
```

Comment:
I'd prefer to show all filters that do not match.

Reply:
I know; I thought of this. However, this requires additional evaluation. Really worth it? Knowing that one filter didn't match is probably enough, and in Marathon / Metronome we just expose "constraints didn't match"